Parse/src/main/java/com/parse/ParseUser.java

```diff
@@ -501,6 +501,18 @@ private void restoreAnonymity(Map<String, String> anonymousData) {
   }
 
   @Override
+  /* package */ void setState(ParseObject.State newState) {
+    // Avoid overwriting session token
```

Comment:
Let's add a little more description here. Something like:

    // Avoid clearing sessionToken when updating the current user's State via ParseQuery result